src/react/fileUrlToFileUrlObject.ts

```diff
@@ -5,32 +5,63 @@ import { getNtrData } from '../shared/ntrData'
 import {
   anyDynamicFilepathPartRegex,
   dynamicFilepathPartsRegex,
+  getCatchAllPathPartKey,
   getDynamicPathPartKey,
+  getOptionalCatchAllPathPartKey,
+  matchAllFilepathPartsRegex,
   optionalMatchAllFilepathPartRegex,
   spreadFilepathPartRegex,
 } from '../shared/regex'
 import type { TRouteBranch } from '../types'
 import { parseUrl } from './parseUrl'
 
+/**
+ * Get the route branch name from a route branch based on the locale or the name if no value for locale is found
+ */
+const getRouteBranchName = (routeBranch: TRouteBranch, locale: string) => {
+  if (!routeBranch?.name && routeBranch?.paths?.[locale]) {
```

Comment:
No, that's actually intended. I forgot to add a comment to clarify that this line omits the root of `routesTree`.

`routesTree` looks like the following on your tests
```js
{
        name: '',
        paths: { default: '', en: 'root' },
        children: [
          { name: 'about', paths: [Object] },
          { name: 'contact', paths: [Object] },
          { name: 'faq', paths: [Object] },
          { name: 'index', paths: [Object] },
          { name: 'auth', paths: [Object], children: [Array] },
          { name: 'catch-all', paths: [Object], children: [Array] },
          { name: 'catch-all-or-none', paths: [Object], children: [Array] },
          { name: 'communities', paths: [Object], children: [Array] },
          { name: 'community', paths: [Object], children: [Array] },
          { children: [Array], name: 'my-account', paths: [Object] },
          { name: 'news', paths: [Object], children: [Array] }
        ]
      }
```

If you don't omit the root you end up with the following error when running the tests:
![CleanShot 2023-10-16 at 16 34 28@2x](https://github.com/hozana/next-translate-routes/assets/13878860/4ea82496-ab0e-4125-8af2-7f12e28fd1c4)

We have to omit `root/` because after `fileUrlToFileUrlObject` returns the pathname `getTranslatedPathPattern` is called and this method maps through the `routeBranch.children` omitting the root and attaching it when the function finish the recursion.
Example:

```js
// fileUrlToUrl.ts
try {
    const { pathname, query, hash } = fileUrlToFileUrlObject({
      fileUrl: url,
      locale,
    })
    // e.g. pathname -> /community/[communityId]/[communitySlug]/statistics

    const { routesTree, defaultLocale } = getNtrData()

    const pathParts = (pathname || '/')
      .replace(/^\/|\/$/g, '')
      .split('/')
      .filter(Boolean)
    // e.g. pathParts -> [ 'community', '[communityId]', '[communitySlug]', 'statistics' ]

    const pathPattern = getTranslatedPathPattern({
      routeBranch: routesTree,
      pathParts,
      locale,
    })
    // e.g. pathPattern -> /root/community/:communityId(\d+){-:communitySlug}/statistics

```
If you don't skip the root `pathPath` will have `[ 'root', 'community', '[communityId]', '[communitySlug]', 'statistics' ]` as value hence the match child logic inside `getTranslatedPathPattern` will fail since is skipping the root and adding it at the end of the recursion as previously stated.

I have came up with a different solution that is adding a `depth` param to `getFileUrlObject` in order to don't return the `routeBranchName` as part of the pathname when you are at depth 0 as that will return the root value.